tests/cypress/e2e/unit_tests/p1_fleet.spec.ts

```diff
@@ -475,3 +475,58 @@ describe('Test Self-Healing on IMMUTABLE resources when correctDrift is enabled'
     }
   )
 });
+
+describe('Test application deployment based on clusterGroup', { tags: '@p1'}, () => {
+  qase(25,
+    it("Fleet-25: Test install single application to the all available clusters in the 'clusterGroup'", { tags: '@fleet-25' }, () => {
+      const repoName = 'default-single-app-cluster-group-25'
+      const key = 'key_env'
+      const value = 'value_prod'
+      const clusterGroupName = 'cluster-group-env-prod'
+      const dsClusterList = ["imported-0", "imported-1"]
+      const bannerMessageToAssert = 'Matches 2 of 3 existing clusters, including "imported-0"'
+
+      cy.accesMenuSelection('Continuous Delivery', 'Clusters');
+      cy.contains('.title', 'Clusters').should('be.visible');
+
+      // Assign label to the clusters 
+      dsClusterList.forEach(
+        (dsClusterName) => {
+          cy.assignClusterLabel(dsClusterName, key, value);
+        }
+      )
+
+      // Create group of cluster consists of same label.
+      cy.clickNavMenu(['Cluster Groups']);
+      cy.contains('.title', 'Cluster Groups').should('be.visible');
+      cy.createClusterGroup(clusterGroupName, key, value, bannerMessageToAssert);
+
+      // Create a GitRepo targeting cluster group created.
+      cy.addFleetGitRepo({ repoName, repoUrl, branch, path, deployToTarget: clusterGroupName });
+      cy.clickButton('Create');
+      cy.checkGitRepoStatus(repoName, '1 / 1');
+
+      // Check application status on both clusters.
+      dsClusterList.forEach(
+        (dsClusterName) => {
+          // Adding wait to load page correctly to avoid interference with hamburger-menu.
+          cy.wait(500);
+          cy.checkApplicationStatus(appName, dsClusterName, 'All Namespaces');
+        }
+      )
+
+      // Remove labels from the clusters.
+      dsClusterList.forEach(
+        (dsClusterName) => {
+          // Adding wait to load page correctly to avoid interference with hamburger-menu.
+          cy.wait(500);
```

Comment:
Did we tried as commented checking a landing locator rather than waits?
Not mandatory removal, just suggestion.

Reply:
We didn't find the landing locator after navigating to `hamburgermenu`...

Reply:
Not in the a few mins meeting, but thought it was going to be tried look for with more time later

Reply:
I approved pr. I leave opt to you to try or merge as it is.

Reply:
I did much investigation on vertical navigation part and after playing with many locators which will be available when it is open and hidden after close...later I found that we will only have to use `cy.wait(500);`. But still I have added extra element check after click happened (if that didn't work as expected, I will remove it.).